docs/agents/models.md

```diff
@@ -27,7 +27,73 @@ ADK primarily uses two mechanisms for model integration:
 
 The following sections guide you through using these methods based on your needs.
 
-## Using Google Gemini Models
+## Google Gemini models
+
+ADK support the Google Gemini family of generative AI models that provide a powerful
+set of models with a wide range of features. ADK provides a number of
+integrations with Gemini models to support features including
+[Code Execution](/adk-docs/tools/built-in-tools/#code-execution),
+[Google Search](/adk-docs/tools/built-in-tools/#google-search),
+[Context caching](/adk-docs/context/caching/),
+[Computer use](/adk-docs/tools/gemini-api/computer-use/)
+and the [Interactions API](#interactions-api).
+
+### Gemini Interactions API {#interactions-api}
+
+<div class="language-support-tag" title="Java ADK currently supports Gemini and Anthropic models.">
+  <span class="lst-supported">Supported in ADK</span><span class="lst-python">Python v1.21.0</span>
+</div>
+
+The Gemini [Interactions API](https://ai.google.dev/gemini-api/docs/interactions)
+provides stateful conversation capabilities, allowing you to chain interactions
+using a `previous_interaction_id` instead of sending the full conversation
+history with each request. Using this feature can be more efficient for long
+conversations.
+
+You can enable the Interactions API by settting the `use_interactions_api=True`
+parameter in the Gemini model configuration, as shown in the following code
+snippet:
+
+```python
+from google.adk.agents.llm_agent import Agent
+from google.adk.models.google_llm import Gemini
+from google.adk.tools.google_search_tool import GoogleSearchTool
+
+root_agent = Agent(
+    model=Gemini(
+        model="gemini-2.5-flash",
+        use_interactions_api=True,  # Enable Interactions API
+    ),
+    name="interactions_test_agent",
+    tools=[
+        GoogleSearchTool(bypass_multi_tools_limit=True),  # Converted to function tool
+        get_current_weather,  # Custom function tool
+    ],
+)
+```
+
+For a complete code sample, see the
+[Interactions API sample](https://github.com/google/adk-python/tree/main/contributing/samples/interactions_api).
+
```

Comment:
It would be nice to include before and after callbacks which utilize the new interaction_id and states.

What handles the status update notifications? 

What handles the resolution of the interaction into the state upon completion of the long-running task?